### PR TITLE
[FIX] save image in base 64

### PR DESCRIPTION
**Pull request by arwazjamal**
https://www.loom.com/share/b6bc398fbda54fea8e1144a3c9b85acc 
This PR creates a canvas from image url and save the image data in base64 so that we wont rely on urls

### CODE REVIEW

**Review by rahulyadav — APPROVED** [review]

File: src/index.js [hunk 1]
```diff
@@ -122,7 +131,56 @@ class SimpleImage {
...
+
+  createFileButton() {
+    const uploadButton = this._make('div', ['cdx-button']);
+    uploadButton.innerHTML = `${this.buttonIcon} Select an Image`;
+    this.uploadFile = this._make('input');
+    this.uploadFile.setAttribute('type', 'file');
+    this.uploadFile.setAttribute('id', 'uploadedFile');
+    this.uploadFile.setAttribute('accept', 'image/*');
+    uploadButton.appendChild(this.uploadFile);
+    uploadButton.addEventListener('click', this.uploadButtonClick);
```
Comment (VarMit24) [inline src/index.js:155]: why we need this listener?

File: src/index.js [hunk 2]
```diff
@@ -122,7 +131,56 @@ class SimpleImage {
...
+    hiddenEl.setAttribute('tabindex', 0);
+    this.wrapper.appendChild(hiddenEl);
+  }
+  uploadButtonClick = () => {
+    this.uploadFile.click();
+  }
+  onSelectFile = () => {
+  const selectedFile = document.getElementById('uploadedFile');
+  const file = selectedFile.files[0];
+  if(file.type === ('image/jpeg') || file.type === ('image/png')) {
```
Comment (VarMit24) [inline src/index.js:169]: can store file types in a list and check if type is present in that or not

File: src/index.js [hunk 3]
```diff
@@ -122,7 +131,56 @@ class SimpleImage {
...
+    this.onDropHandler(file).then(data => {
+      this.data = data;
+    });
+    this._createImage();
+  } else {
+    this.api.notifier.show({
+      message: 'Can not upload an image, try another',
```
Comment (VarMit24) [inline src/index.js:176]: Not a supported type, please try other

File: src/index.js [hunk 4]
```diff
@@ -122,7 +131,56 @@ class SimpleImage {
...
+        this.uploadFile.click();
+      }
+    }
+    return this.wrapper;
+  }
+
+  createFileButton() {
+    const uploadButton = this._make('div', ['cdx-button']);
+    uploadButton.innerHTML = `${this.buttonIcon} Select an Image`;
+    this.uploadFile = this._make('input');
```
Comment (VarMit24) [inline src/index.js:150]: can we rename this uploadFileInput

File: src/index.js [hunk 5]
```diff
@@ -122,7 +131,56 @@ class SimpleImage {
...
+    this.uploadFile.setAttribute('type', 'file');
+    this.uploadFile.setAttribute('id', 'uploadedFile');
+    this.uploadFile.setAttribute('accept', 'image/*');
+    uploadButton.appendChild(this.uploadFile);
```
Comment (VarMit24) [inline src/index.js:154]: uploadFileButton

File: src/index.js (hunk 1 above)
Comment (arwazjamal) [reply src/index.js:155]: now i have hidden the file input and showed a fileupload button, user clicks this button on which events for file-input is triggered programatically

**Pull request merged**